projects/core/src/site-theme/facade/site-theme.service.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, OnDestroy, inject } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { filter, map, mergeMap, take, tap } from 'rxjs/operators';
+
+import { Observable, Subscription } from 'rxjs';
+
+import { SiteTheme } from '../../model/misc.model';
+import { SiteThemeConfig } from '../config/site-theme-config';
+import { isNotNullable } from '../../util/type-guards';
+import { StateWithSiteTheme } from '../store/state';
+import { SiteThemeSelectors } from '../store/selectors';
+import { SiteThemeActions } from '../store/actions';
+
+@Injectable()
+export class SiteThemeService implements OnDestroy {
+  protected _isInitialized = false;
+  protected store = inject(Store<StateWithSiteTheme>);
+  protected config = inject(SiteThemeConfig);
+  protected subscription = new Subscription();
+
+  ngOnDestroy(): void {
+    this.subscription.unsubscribe();
+  }
+  // init(className: string): Observable<string> {
+
+  // }
+
+  getAll(): Observable<SiteTheme[]> {
+    return this.store.pipe(
+      select(SiteThemeSelectors.getAllSiteThemes),
+      tap((themes) => {
+        if (!themes) {
+          this.store.dispatch(new SiteThemeActions.LoadSiteThemes());
+        }
+      }),
+      filter(isNotNullable)
+    );
+  }
+
+  /**
+   * Gets the className of the active theme.
+   */
+  getActive(): Observable<string> {
+    return this.store.pipe(
+      select(SiteThemeSelectors.getActiveSiteTheme),
+      filter(isNotNullable)
+    );
+  }
+
+  /**
+   * Sets the active theme className.
+   */
+  setActive(className: string): Observable<void> {
+    return this.isValidTheme(className).pipe(
+      filter(Boolean),
+      mergeMap(() =>
+        this.store.pipe(select(SiteThemeSelectors.getActiveSiteTheme), take(1))
+      ),
+      tap((activeTheme) => {
+        if (activeTheme !== className) {
+          this._isInitialized = true;
+          this.store.dispatch(
+            new SiteThemeActions.SetActiveSiteTheme(className)
+          );
+        }
+      }),
+      map(() => undefined),
+      take(1)
+    );
+  }
```

Comment:
`major:` let's clean up this method a bit
```suggestion
  setActive(className: string): void {
    this.isValidTheme(className)
      .pipe(filter(Boolean), take(1))
      .subscribe(() => {
        this.store.dispatch(new SiteThemeActions.SetActiveSiteTheme(className));
      });
    this._isInitialized = true;
  }
```
- use `take(1)` to handle unsubscription
- make a subscription here to make usage of this method cleared
- remove if statement `if (activeTheme !== className)` - I think this micro improvement provides more code complexity than the real value.
- let's set `isInitialized` outside the stream. We have a race condition between `this.siteThemePersistenceService.initSync()` and `this.setFallbackValue()` but I couldn't find the reason. Let's add `TODO` with a link to the separate ticket to investigate the issue and found a better way of checking whether theme is initialized, but let's not block this feature.